implmented the query functionality into the api/articles endpoint and…

Cralus opened the pull request:
Implemented using queries for the /api/articles endpoint and 400 errors for bad queries to sort by and order by

Cralus committed:
implmented the query functionality into the api/articles endpoint and associated error testing

cjplatten commented on __tests__/app.test.js:
```diff
@@ -324,6 +324,44 @@ describe('POST /api/articles/:article_id/comments', () => {
+                ascending: true,
+            });
+        });
+    });
```
It would be good to see a separate test demonstrating the sort_by, at the moment this test is using the default

cjplatten commented on __tests__/app.test.js:
```diff
@@ -324,6 +324,44 @@ describe('POST /api/articles/:article_id/comments', () => {
+        });
+
+});      
+})
```
There are few additional tests it would be good to see for the topic! What if the topic given is invalid or not found? Or if there are no articles for a topic that exists?

cjplatten commented on controllers/articles.js:
```diff
@@ -26,8 +27,12 @@ exports.patchArticlesById = (req, res, next) => {
-  fetchArticles().then((articles) => {
+exports.getArticles = (req, res, next) => { 
+  const query = req.query;
+  console.log(query)
```
Console.log!

cjplatten reviewed (changes requested):
A few comments to address, I like the handling so far!

Cralus committed:
added error handling for when the query for topics does not match any existing topics

Cralus committed:
removed console logs

Cralus commented:
I've refactored the code, added a function in the topics model to check if the topic exists then used that function in the articles controller to respond with the correct error if the requested topic exists or not, topics without articles return a 200 as its not an error. and removed all the console logs! also added a new test for sort by where its not using the default

dappernerddesigns reviewed (approved):
Nicely handled, good refactor.
